pyrefly/lib/lsp/non_wasm/server.rs

```diff
@@ -1167,14 +1178,34 @@ impl Server {
         let publish = |transaction: &Transaction| {
             let mut diags: SmallMap<PathBuf, Vec<Diagnostic>> = SmallMap::new();
             let open_files = self.open_files.read();
+
+            // Initialize diagnostic entries for open files
             for x in open_files.keys() {
```

Comment:
we probably don't want to do this. if we have the entry.or_default() behavior below I don't think it does anything